src/main/java/seedu/duke/ui/AppUi.java

```diff
@@ -105,12 +105,41 @@ public void deleteEntry(HashMap<String, String> commandArguments) {
         deleteCommand.execute(financialList);
     }
 
+    /**
+     * This method helps execute the appropriate command based on the "argument"
+     * provided in the commandArguments HashMap. If the argument is "expense", it will
+     * execute the SeeAllExpensesCommand. If the argument is "income", it will execute
+     * the SeeAllIncomesCommand. If no argument or an unknown argument is provided,
+     * it defaults to executing SeeAllEntriesCommand to list all entries.
+     *
+     * @param commandArguments A HashMap containing the command argument with the key "argument".
+     *                         The value can be "expense", "income", or null/empty for listing all entries.
+     */
+    public void listHelper(HashMap<String, String> commandArguments) {
+        String type = commandArguments.get("argument");
+
+        if (type != null) {
+            if (type.equals("expense")) {
+                SeeAllExpensesCommand seeAllExpensesCommand = new SeeAllExpensesCommand();
+                seeAllExpensesCommand.execute(financialList);
+            } else if (type.equals("income")) {
+                SeeAllIncomesCommand seeAllIncomesCommand = new SeeAllIncomesCommand();
+                seeAllIncomesCommand.execute(financialList);
+            } else {
+                System.out.println("Unknown argument: " + type);
+            }
+        } else {
+            SeeAllEntriesCommand seeAllEntriesCommand = new SeeAllEntriesCommand();
+            seeAllEntriesCommand.execute(financialList);
+        }
+    
```

Comment:
There seems to be a missing close bracket here which is causing the program to be unable to build